### PR TITLE
Daemon monitoring + Dockerfile

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
Sets up collectors which exposes metrics at `$API/debug/metrics/prometheus` (as in go-ipfs) that can be consumed by Prometheus.

Ref: #804

### CODE REVIEW

File: src/http-api/routes/debug.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const register = require('prom-client').register
+const client = require('prom-client')
+
+// Endpoint for handling debug metrics
+module.exports = (server) => {
+  const api = server.select('API')
+  const gauge = new client.Gauge({ name: 'number_of_peers', help: 'the_number_of_currently_connected_peers' })
+
+  setInterval(() => {
```
Comment (VictorBjelkholm) [inline src/http-api/routes/debug.js]: Not sure about this approach. Guess we have two options, 1) (which is implemented here) is to once every X seconds, check the number of peers and save it or 2) when a request happens at `/debug/metrics/prometheus`, then check the number of peers and respond.

@diasdavid what you think?

**Comment by VictorBjelkholm** [comment]
Missing: 

- [ ] Use webrtc-star by installing it globally
- [ ] Information about the endpoint in the documentation somewhere

File: src/http-api/index.js [hunk 2]
```diff
@@ -23,6 +26,11 @@ function HttpApi (repo, config) {
   this.log = debug('jsipfs:http-api')
   this.log.error = debug('jsipfs:http-api:error')
 
+  // Setup debug metrics collection
+  const collectDefaultMetrics = prometheusClient.collectDefaultMetrics
+  collectDefaultMetrics({ timeout: 5000 })
+  prometheusGcStats(prometheusClient.register)()
```
Comment (diasdavid) [inline src/http-api/index.js]: Does this mean that every js-ipfs node will "phone home"?

File: src/http-api/routes/debug.js (hunk 1 above)
Comment (diasdavid) [inline src/http-api/routes/debug.js]: let's calculate per request

File: src/http-api/index.js (hunk 2 above)
Comment (VictorBjelkholm) [inline src/http-api/index.js]: @diasdavid "phone home" as in send the data somewhere? No, it'll only be in the local node, available to collect when hitting the endpoint. The data is not sent somewhere.

Comment (diasdavid) [inline src/http-api/index.js]: got it. Still, we should not add stuff that is not necessary by default. Can these be in a separate file that doesn't get loaded unless a env var is set?

Comment (VictorBjelkholm) [inline src/http-api/index.js]: Sure, that makes sense, although it makes the usage different from go-ipfs who exposes it by default. But that makes sense, I'll move the dependencies to optionalDependencies as well then.

Comment (diasdavid) [inline src/http-api/index.js]: Thank you! :)

**Comment by diasdavid** [comment]
@VictorBjelkholm how close are we from this?

**Comment by VictorBjelkholm** [comment]
@diasdavid just resolved a conflict, once CI is passing, is ready to be merged 👍

**Comment by diasdavid** [comment]
rebase master onto this branch. it will solve the issues you are seeing.

**Comment by diasdavid** [comment]
With the work of @harshjv and @ya7ya, we will have a gateway as well and from what I've discussed with Lars, we will be running some gateways as js-ipfs nodes for testing :)

**Pull request merged**